src/linux/linux_io.c

```diff
@@ -335,6 +351,50 @@ static int create_server_socket_all_interfaces(int port)
 	return -1;
 }
 
+static int create_server_unix_domain_socket(const char *pPath)
+{
+	int listen_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+	if (unlikely(listen_fd < 0)) {
+		log_err("Could not create listen socket!\n");
+		return -1;
+	}
+
+	static const int reuse_on = 1;
+	if (unlikely(setsockopt(listen_fd, SOL_SOCKET, SO_REUSEADDR, &reuse_on,
+							sizeof(reuse_on)) < 0)) {
+		log_err("Could not set %s!\n", "SO_REUSEADDR");
+		goto error;
+	}
+
+	if (unlikely(set_fd_non_blocking(listen_fd) < 0)) {
+		log_err("Could not set %s!\n", "O_NONBLOCK");
+		goto error;
+	}
+
+	struct sockaddr_un serveraddr;
+	memset(&serveraddr, 0, sizeof(serveraddr));
+	serveraddr.sun_family = AF_UNIX;
+	strncpy(serveraddr.sun_path, pPath, sizeof(serveraddr.sun_path)-1);
```

Comment:
Should be
strncpy(serveraddr.sun_path, pPath, sizeof(serveraddr.sun_path) - 1);

Reply:
done